test/test.js

```diff
@@ -101,4 +101,42 @@ describe('enhanceWithClickOutside', () => {
       'click', enhancedComponent.handleClickOutside, true
     )
   });
+
+  it('should always call the handleClickOutside if wrapped components render returns null', () => {
+    const clickOutsideSpy = expect.createSpy();
+
+    const WrappedComponent = React.createClass({
+      displayName: 'NullComponent',
+
+      handleClickOutside() {
+        clickOutsideSpy();
+      },
+
+      render() {
+        return null;
+      }
+    });
+    const EnhancedComponent = enhanceWithClickOutside(WrappedComponent);
+
+    const RootComponent = React.createClass({
+      render() {
+        return (
+            <div>
+              <EnhancedComponent ref="enhancedComponent" />
+            </div>
+        );
+      }
+    })
+
+    const rootComponent = React.render(<RootComponent />, document.body);
+
+    // We shouldn't TypeError when we try to call handleClickOutside
+    expect(() => {
+      rootComponent.refs.enhancedComponent.handleClickOutside();
+    }).toNotThrow("Cannot read property 'contains' of null");
```

Comment:
This error message is V8 specific. Replacing this with `toNotThrow(TypeError)` would be more robust. Also if you fix this please squash/fixup the commit.